File: MODUL-2/SOAL 2/PRAK202-2410817120017-RIKA-FAULIANA-RAHMI.c
#include <stdio.h>

void format_angka(float angka) {
    int integer_part = (int)angka;
    float decimal_part = angka - integer_part;

    if (decimal_part == 0) {
        // Jika angka adalah bilangan bulat, tampilkan tanpa desimal
        printf("%d", integer_part);
    } else if ((decimal_part * 10) == (int)(decimal_part * 10)) {
        // Jika angka memiliki satu angka desimal yang tidak nol
        printf("%.1f", angka);
    } else {
        // Jika angka memiliki lebih dari satu angka desimal
        printf("%.2f", angka);
    }
}


int main() {
    float nilai1, nilai2, hasil;

    // Input nilai pertama
    printf("Masukkan Nilai Pertama: ");
    scanf("%f", &nilai1);

    // Input nilai kedua
    printf("Masukkan Nilai Kedua: ");
    scanf("%f", &nilai2);

    // Menghitung hasil penjumlahan
    hasil = nilai1 + nilai2;

    // Output dengan format yang sesuai
    printf("Hasil dari penjumlahan nilai pertama \"");
    format_angka(nilai1);
    printf("\" dan nilai kedua \"");
    format_angka(nilai2);
    printf("\" adalah \"%.2f\"\n", hasil);  // Hasil selalu dua angka di belakang koma

    return 0;
}

File: MODUL-2/SOAL 2/PRAK202-2410817120017-RIKA-FAULIANA-RAHMI.py
# Fungsi untuk format angka sesuai dengan input
def format_angka(angka, is_result=False):
    if is_result:
        return f"{angka:.2f}"  # Menampilkan hasil dengan dua angka di belakang koma
    elif angka.is_integer():
        return f"{int(angka)}"  # Menghilangkan ".00" jika bilangan bulat
    elif (angka * 10).is_integer():
        # Jika hanya memerlukan satu angka desimal, tampilkan dengan satu angka di belakang koma
        return f"{angka:.1f}"
    else:
        return f"{angka:.2f}"   # Jika memiliki lebih dari satu angka desimal, tampilkan dua angka di belakang koma

# Input nilai pertama
nilai1 = float(input("Masukkan Nilai Pertama: "))

# Input nilai kedua
nilai2 = float(input("Masukkan Nilai Kedua: "))

# Menghitung hasil penjumlahan
hasil = nilai1 + nilai2

# Output dengan format sesuai input, dan hasil selalu dua angka di belakang koma
print(f'Hasil dari penjumlahan nilai pertama "{format_angka(nilai1)}" dan nilai kedua "{format_angka(nilai2)}" adalah "{format_angka(hasil, is_result=True)}"')

